src/Injector.php

```diff
@@ -39,20 +38,12 @@ class Injector implements InjectorInterface
      */
     protected $autoCreateWhiteList = [];
 
-    /**
-     * @var ParameterInspector
-     */
-    private $inspector;
+    private ClassInspectorInterface $classInspector;
 
-    /**
-     *
-     * @param ContainerInterface $container
-     * @param ParameterInspector $inspector
-     */
-    public function __construct(ContainerInterface $container, ParameterInspector $inspector)
+    public function __construct(ContainerInterface $container, ClassInspectorInterface $classInspector)
```

Comment:
🍹 can probably use constructor property promotion here to remove the boilerplate:
```suggestion
    public function __construct(
        private ContainerInterface $container,
        private ClassInspectorInterface $classInspector,
    )
```